packages/valory/skills/decision_maker_abci/behaviours/sampling.py

```diff
@@ -95,6 +125,13 @@ def _sample(self) -> Optional[int]:
         """Sample a bet, mark it as processed, and return its index."""
         available_bets = list(filter(self.processable_bet, self.bets))
 
+        # store the bet selection stats for each processable bet at the time of sampling
+        for bet in available_bets:
+            self.shared_state.bet_selection_stats[bet.id] = {
+                "scaled_liquidity_measure": bet.scaledLiquidityMeasure,
+                "outcome_token_amounts": bet.outcomeTokenAmounts,
+            }
```

Comment:
Please convert the nested dict to a dataclass to avoid human error.

Reply:
ok